### PR TITLE
Master

### PR DESCRIPTION
**Pull request by fare1990**
Added new command to set autocomplete templates.
Added simple security in new command execution

### CODE REVIEW

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteHandler.java [hunk 1]
```diff
@@ -20,11 +20,12 @@
 
     private final BotApi botApi;
     private final Map<String, String[]> autocompletes = new HashMap<>();
+    private final String pattern;
 
     @Autowired
     public AutocompleteHandler(BotApi botApi, AutocompleteConfig config) {
         this.botApi = botApi;
-
+        this.pattern = Pattern.compile("[^\\/\\n]+\\/([^\\/\\n]+\\/)*[^\\/\\n]+").toString();
```
Comment (denzelby) [inline telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteHandler.java]: .toString()? Why? You should compile and cache pattern in field for further use.

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteAddCommand.java [hunk 2]
```diff
@@ -31,12 +31,12 @@ public void handleCommand(Update update, Long chatId, String argument) {
             return;
         }
 
-        if(!handler.addTemplate(argument.trim())){
-            botApi.sendMessage(chatId, "Wrong template, try again.");
+        if(argument==null || !handler.addTemplate(argument.trim())){
+            botApi.sendMessage(chatId, "wrong template, try again.");
```
Comment (denzelby) [inline telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteAddCommand.java]: As you can see, you rewrote last commit with your changes. Even if so, could you please make these changes manually to not throw previous changes away?

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteAddCommand.java [hunk 3]
```diff
@@ -31,12 +31,12 @@ public void handleCommand(Update update, Long chatId, String argument) {
             return;
         }
 
-        if(!handler.addTemplate(argument.trim())){
-            botApi.sendMessage(chatId, "Wrong template, try again.");
+        if(argument==null || !handler.addTemplate(argument.trim())){
```
Comment (denzelby) [inline telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteAddCommand.java]: Invocation of methods with side effects in "if" closure is a bad practice.

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteHandler.java [hunk 4]
```diff
@@ -20,11 +20,12 @@
 
     private final BotApi botApi;
     private final Map<String, String[]> autocompletes = new HashMap<>();
+    private final Pattern pattern;
 
     @Autowired
     public AutocompleteHandler(BotApi botApi, AutocompleteConfig config) {
         this.botApi = botApi;
-
+        this.pattern = Pattern.compile("[^\\/\\n]+\\/([^\\/\\n]+\\/)*[^\\/\\n]+");
```
Comment (nchereva) [inline telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteHandler.java]: How about constants or move it into config file?

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteHandler.java [hunk 5]
```diff
@@ -39,15 +40,13 @@ public boolean addTemplate(String phraseTemplate) {
     }
 
     private boolean isValidTemplate(String argument) {
-        if (argument == null ||
-                !argument.contains("/") ||
-                argument.substring(0, argument.indexOf('/')).length() + 1 >= argument.length()) {
-
+        if (!pattern.matcher(argument).matches()) {
```
Comment (nchereva) [inline telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/autocomplete/AutocompleteHandler.java]: Could be simplified into single `return` statement:
```java
return pattern.matcher(argument).matches();
```

**Comment by denzelby** [comment]
Please fix code review issues and create new PR compatible with latest version.

**Pull request closed without merging**